libsolidity/ast/Types.cpp

```diff
@@ -151,6 +151,33 @@ util::Result<TypePointers> transformParametersToExternal(TypePointers const& _pa
 	return transformed;
 }
 
+BoolResult validSizeForLocation(bigint const& _size, DataLocation _loc)
+{
+	if (_loc == DataLocation::Storage && _size >= (bigint(1) << 256))
+		return BoolResult::err("Type too large for storage.");
+	if (_loc == DataLocation::Memory && _size >= numeric_limits<unsigned>::max())
```

Comment:
I've always disliked ``numeric_limits<unsigned>::max()``... It depends on what size ``unsigned`` is in the given platform and that has really no bearing on the size of solidity memory types.
Maybe just change that to ``(bigint(1) << 64)`` or ``numeric_limits<uint64_t>::max()`` or something like that (I'm aware that there's a difference of one between them ;-)).

Reply:
See previous discussion in https://github.com/ethereum/solidity/pull/9746#discussion_r484426796

I agree that it should be changed but `unsigned`  is used all over the place in `Type`, including other size checks. I think it doesn't make sense to change just this one place. We should change them all consistently.

Reply:
Sure. But just a side remark. As far as I remember, some of these checks were meant to avoid ICEs. So in theory sizes _could be_ related to a flavor of solc compiler too. Nevertheless, let's proceed as suggested. (BTW `int64_t` seems to be [optional](https://en.cppreference.com/w/cpp/types/integer).)